static/app/views/performance/transactionSummary/transactionEvents/eventsTable.tsx

```diff
@@ -326,6 +349,37 @@ class EventsTable extends Component<Props, State> {
         }
         return col;
       });
+
+    const joinCustomData = ({data}: TableData) => {
+      if (this.state.attachments.length) {
+        const {projectId} = this.props;
+        const attachmentsWithUrl = this.state.attachments.map(attachment => ({
+          ...attachment,
+          url: `/api/0/projects/${organization.slug}/${projectId}/events/${attachment.event_id}/attachments/${attachment.id}/`,
+        }));
+        const eventIdMap = {};
+        data.forEach(event => {
+          event.attachments = [] as any;
+          eventIdMap[event.id] = event;
+        });
+        attachmentsWithUrl.forEach(attachment => {
+          const eventAttachments = eventIdMap[attachment.event_id]?.attachments;
+          if (eventAttachments) {
+            eventAttachments.push(attachment);
+          }
+        });
+      }
+    };
```

Comment:
Totally fine, but it's sometimes confusing to mutate data like this. Another way that might be simpler is to use lodash. e.g., `attachmentsByEvent = groupBy(state.attachments, 'event_id')` and pass that to the renderer by attaching it to `data`. Then in the renderer look up the attachments by the event ID. Less code, simpler types, etc.